### PR TITLE
Cosmetic fixes to documentation in `BaseStorage`.

### PR DESCRIPTION
**Pull request by hvy**
Updates the documentation of the `BaseStorage`. Changes target English grammar and simplification, style and formatting, and some consistencies. In short, they're mostly cosmetic. 

Refer to https://github.com/optuna/optuna/pull/1175#pullrequestreview-412446405 for more context.

### CODE REVIEW

File: optuna/storages/base.py [hunk 1]
```diff
@@ -21,88 +21,87 @@ class BaseStorage(object, metaclass=abc.ABCMeta):
 
     This class is not supposed to be directly accessed by library users.
 
-    Storage classes abstract a backend database and provide library internal interfaces to
-    read/write history of studies and trials.
+    A storage class abstract a backend database and provide library internal interfaces to
```
Comment (ytsmiling) [inline optuna/storages/base.py:24]: nit:
```suggestion
    A storage class abstracts a backend database and provide library internal interfaces to
```

Comment (hvy) [reply optuna/storages/base.py:24]: Ops, thanks for the catch. Let me apply your change.

Comment (hvy) [reply optuna/storages/base.py]: "provide" should probably be fixed too to "provides". Let me fix it.

Comment (hvy) [reply optuna/storages/base.py]: Fixed.

**Comment by hvy** [comment]
Thank you for the fast review. I pushed some minors changes including your suggestion. PTAL.

**Pull request merged**